### PR TITLE
Move docs and gallery deployment to Cirrus

### PR DESCRIPTION
**Pull request by gspencergoog**
This moves the docs generation and gallery upload to Cirrus CI, which is another step in turning off
Travis.

### CODE REVIEW

File: dev/bots/cirrus_setup.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  gradle -v
+  set +x
+}
+
+echo "Flutter SDK directory is: $PWD"
+
+if [[ -n "$CIRRUS_CI" && "$OS_NAME" == "linux" && "$SHARD" == "deploy_gallery" ]]; then
+  setup_android
+fi
+
```
Comment (tvolkert) [inline dev/bots/cirrus_setup.sh]: Do we no longer need to set up the Google Cloud SDK for docs?

https://github.com/flutter/flutter/blob/82415e640001fd2934cb1cc9ceed902cea4a2025/dev/bots/travis_setup.sh#L10

File: dev/bots/cirrus_setup.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# It must be run from the root of the Flutter repo.
+
+# Collects log output in a tmpfile, but only prints it if the command fails.
+function log_on_fail() {
+  local COMMAND="$@"
+  local TMPDIR="$(mktemp -d)"
+  local TMPFILE="$TMPDIR/command.log"
+  ("$@" > "$TMPFILE" 2>&1 && echo "'$COMMAND' succeeded.") || \
+    (EXIT=$? && cat "$TMPFILE" 1>&2 && echo "FAIL: '$COMMAND' exited with code $EXIT" 1>&2 || false)
+  # If we got here, we succeeded, so remove the tmpfile.
```
Comment (tvolkert) [inline dev/bots/cirrus_setup.sh]: Shouldn't we remove the tmp file regardless?

**Review by tvolkert — APPROVED** [review]
LGTM

File: .cirrus.yml [hunk 3]
```diff
@@ -3,26 +3,26 @@ container:
...
   matrix:
     - name: docs
       env:
         SHARD: docs
+        # For uploading docs to Firebase
+        FIREBASE_TOKEN: ENCRYPTED[a40fcb68452b92254b7f532a460529b5e4ca51d6d338a8fae8db9db832ad3c2a7efb962e257de79686a9345f4a18661a]
       docs_script: ./dev/bots/docs.sh
+    - name: deploy_gallery
+      env:
+        SHARD: deploy_gallery
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply .cirrus.yml]: Yes, it should be set.  Not all of the scripts assume that (the docs script doesn't), but they should be able to.

In this case, it is set to `linux`, from the `env:` section at the task level, so we don't need to set it in the matrix here.

**Comment by fkorotkov** [comment]
@gspencergoog after this change `docs` shard is hanging with:

```
./dev/bots/docs.sh: line 44: firebase: command not found
Error: Unable to deploy documentation to firebase. Retrying in five seconds...
```

**Comment by fkorotkov** [comment]
fixed in #19999

**Pull request merged**